src/navigation/_CollapsingNavigationMenu.jsx

```diff
@@ -125,65 +135,136 @@ const CollapsingNavigationMenu = ({ selectedPath = undefined, menuItems, onSelec
+      event.preventDefault();
+      cursor.current = visibleNodes.current.length - 1;
+      setCurrentNodeId(visibleNodes.current[cursor.current].id);
+    } else if (event.nativeEvent.keyCode >= KeyCode.KEY_A && event.nativeEvent.keyCode <= KeyCode.KEY_Z) {
+      event.preventDefault();
+      const char = String.fromCharCode(event.nativeEvent.keyCode);
+      findNode(char);
     }
   };
 
-  const renderMenuItems = (currentMenuItem, firstLevel) => {
+  const renderMenuItems = (currentMenuItem, parent = '', firstLevel = false) => {
     if (!currentMenuItem) {
       return undefined;
     }
 
     return currentMenuItem.map((item) => {
       const itemIsOpen = openKeys[item.path];
       const itemHasChildren = item.childItems !== undefined;
+      const id = item.name.split(' ').join('-');
       let isSelected = false;
       let selectedRef;
 
-      if (selectedPath === item.path) {
+      if (selectedPath === item.path || currentNodeId === id) {
         isSelected = true;
         selectedRef = selectedItem;
       }
 
+      visibleNodes.current.push({ id, parent });
       const menuItemClassNames = classNames(
         cx([
           'item',
           { 'is-selected': isSelected },
         ]),
       );
 
+      const optionalAttributes = itemHasChildren ? { 'aria-expanded': itemIsOpen, 'aria-haspopup': true } : {};
+
       return (
         <React.Fragment key={item.path}>
           <div className={!firstLevel ? cx('indent') : null}>
             <div
               className={menuItemClassNames}
               tabIndex="0"
-              role="link"
-              id={item.name.split(' ').join('-')}
-              aria-haspopup={itemHasChildren}
+              role="treeitem"
+              id={id}
               onKeyDown={event => handleKeyDown(event, item)}
               onClick={event => handleOnClick(event, item)}
               onBlur={enableFocusStyles}
               onMouseDown={disableFocusStyles}
               data-focus-styles-enabled
               ref={selectedRef}
+              {...optionalAttributes}
             >
               {itemHasChildren ? <span className={cx('disclosure')}>{ itemIsOpen ? <IconCaretDown className={cx('caret')} /> : <IconCaretRight className={cx('caret')} />}</span> : null}
               {item.name}
             </div>
-            {itemIsOpen ? renderMenuItems(item.childItems) : null}
+            {itemIsOpen ? renderMenuItems(item.childItems, id) : null}
           </div>
         </React.Fragment>
       );
     });
   };
 
   return (
-    <div className={cx('collapsing-navigation-menu', theme.className)} id="terra-dev-site-nav-menu" tabIndex="-1">
-      {menuItems ? renderMenuItems(menuItems[0].childItems, true) : undefined}
+    <div className={cx('collapsing-navigation-menu', theme.className)} id="terra-dev-site-nav-menu" tabIndex="-1" role="tree">
+      {menuItems ? renderMenuItems(menuItems[0].childItems, '', true) : undefined}
     </div>
   );
 };
```

Comment:
The keyboard focus and mouse focus are able to get out of sync:

![image](https://user-images.githubusercontent.com/6720991/85791502-899a7580-b6f7-11ea-83fc-be29155b8b26.png)


Here the Nested Secondary Nav Test has been selected and is rendered, but dashed outline focus remained on the Themed tab